shared/api/update.go

```diff
@@ -169,66 +170,6 @@ type Update struct {
 	Status UpdateStatus `json:"update_status" yaml:"update_status"`
 }
 
-type UpdateFileComponent string
-
-const (
-	UpdateFileComponentOS               UpdateFileComponent = "os"
-	UpdateFileComponentIncus            UpdateFileComponent = "incus"
-	UpdateFileComponentMigrationManager UpdateFileComponent = "migration-manager"
-	UpdateFileComponentOperationsCenter UpdateFileComponent = "operations-center"
-	UpdateFileComponentDebug            UpdateFileComponent = "debug"
-)
-
-var updateFileComponents = map[UpdateFileComponent]struct{}{
-	UpdateFileComponentOS:               {},
-	UpdateFileComponentIncus:            {},
-	UpdateFileComponentMigrationManager: {},
-	UpdateFileComponentOperationsCenter: {},
-	UpdateFileComponentDebug:            {},
-}
-
-func (u UpdateFileComponent) String() string {
-	return string(u)
-}
-
-// MarshalText implements the encoding.TextMarshaler interface.
-func (u UpdateFileComponent) MarshalText() ([]byte, error) {
-	return []byte(u), nil
-}
-
-// UnmarshalText implements the encoding.TextUnmarshaler interface.
-func (u *UpdateFileComponent) UnmarshalText(text []byte) error {
-	_, ok := updateFileComponents[UpdateFileComponent(text)]
-	if !ok {
-		return fmt.Errorf("%q is not a valid update file component", string(text))
-	}
```

Comment:
This check is missing in https://github.com/lxc/incus-os/blob/main/incus-osd/api/images/update_file_component.go#L41-L46, so using the the file components from incus-os is less strict. This feels a bit [YOLO](https://en.wikipedia.org/wiki/YOLO_(aphorism)) to me, since we basically accept anything for the file component and happily store it to the DB and also return it again. So the whole purpose of having a type for `UpdateFileComponent` is use less. It could also well be just a `string`.